.github/workflows/test-library.yml

```diff
@@ -188,7 +188,7 @@ jobs:
     - name: Set the expected dist artifact names
       id: artifact-name
       run: |
-        print('::set-output name=sdist::proxy.py-${{
+        print('::set-output name=sdist::proxy_py-${{
```

Comment:
@webknjaz Apologies for tagging you again, but wanted to highlight this change as I am unsure what upstream changes might have caused this.  Somehow generated `.tar.gz` name has suddenly changed

Reply:
@abhinavsingh no need to apologize :) This update has been widely noticed across many projects lately. `setuptools` made updates to the sdist base name normalization code in one of the recent releases which resulted in underscores appearing in place of `.` or `-`. This would've been more stable if `setuptools` was pinned for the build env (I don't think I implemented this in my earlier PRs but I tend to do this in my other projects).

Here's the corresponding change log entry upstream: https://setuptools.pypa.io/en/latest/history.html#v69-3-0.

Reply:
@abhinavsingh while I'm here, I also noticed the use of `::set-output` GHA workflow command syntax which has been deprecated for a while. The new way is appending to a file path provided to by the `${GITHUB_OUTPUT}` environment variable. For example: https://github.com/cherrypy/cheroot/blob/3591a1c/.github/workflows/ci-cd.yml#L309-L327.

Reply:
@webknjaz You are a charm.  Thank you for providing links and insights, I'll dig into them 🙏